integrationtest/mapper/pojo-base/pom.xml

```diff
@@ -59,5 +59,40 @@
         </plugins>
     </build>
 
+    <profiles>
+        <profile>
+            <id>testWithJdk8</id>
+            <activation>
+                <property>
+                    <name>java-version.test.release</name>
+                    <value>8</value>
+                </property>
+            </activation>
+            <properties>
+                <!--
+                We want to skip -noparameter compilation when testing against JDK < 17 as tests for this execution are
+                for 17+ versions only.
+                -->
+                <maven.compiler.testSources.noParameterCompilation.skip>true</maven.compiler.testSources.noParameterCompilation.skip>
+            </properties>
+        </profile>
+
+        <profile>
+            <id>testWithJdk11</id>
+            <activation>
+                <property>
+                    <name>java-version.test.release</name>
+                    <value>11</value>
+                </property>
+            </activation>
+            <properties>
+                <!--
+                We want to skip -noparameter compilation when testing against JDK < 17 as tests for this execution are
+                for 17+ versions only.
+                -->
+                <maven.compiler.testSources.noParameterCompilation.skip>true</maven.compiler.testSources.noParameterCompilation.skip>
+            </properties>
+        </profile>
+    </profiles>
```

Comment:
Shouldn't this be in the root pom, like (almost) everything else related to this property (`maven.compiler.testSources.noParameterCompilation.skip`)?

Reply:
Ah, I see, that's because you override something defined in this POM. There are ways to control this from the root POM anyway, let me check...

Reply:
So, I *think* we could do something similar to what we do for ORM tests and DB profiles:

https://github.com/hibernate/hibernate-search/blob/684018b3b17f004ae1b3aad3091fda54ba118c8e/integrationtest/mapper/orm/pom.xml#L15-L17

https://github.com/hibernate/hibernate-search/blob/aaa193330ced3a5858daa00a940de1d4ece9f5e2/parents/integrationtest/pom.xml#L695-L698

Basically in your case, I think you would:

* revert this change
* introduce a different property, e.g. `java-version.test.java17.noParameters.skip`, to control whether the maven-compiler execution is skipped
* set `java-version.test.java17.noParameters.skip` to `${maven.compiler.testSources.noParameterCompilation.skip}` by default (in the properties of the root POM)
* force `java-version.test.java17.noParameters.skip` to `false` in the JDK8/JDK11 profiles

Reply:
I'll give it a try a bit later, have to step out -_-